### PR TITLE
Add E2E tests for multi-APIs

### PR DESCRIPTION
**Pull request by rinx**
Signed-off-by: Rintaro Okamura <rintaro.okamura@gmail.com>

### Description:

This PR contains:

- Add E2E tests for multi-APIs
- Use t.CleanUp for calling teardown
- Split operation.go into several files

### Related Issue:

nothing

### How Has This Been Tested?:

nothing

### Environment:

- Go Version: 1.16.6
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.12.3

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [X] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [X] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [X] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [ ] I have added tests and benchmarks to cover my changes.
- [ ] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by kpango — DISMISSED** [review]
LGTM

File: tests/e2e/operation/multi.go [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+		})
+	}
+
+	req := &payload.Search_MultiRequest{
+		Requests: reqs,
+	}
+	_, err = client.MultiSearch(ctx, req)
+	if err != nil {
+		return err
+	}
```
Comment (kevindiu) [inline tests/e2e/operation/multi.go:54]: I suggest to check the result length and also the distance contains 0 (exact result). 
Same as below functions.
```suggestion
	res, err = client.MultiSearch(ctx, req)
	if err != nil {
		return err
	}
	if len(res.Results) != len(ds.Test) {
	        return errors.New("response length does not match with dataset")
	}
	if res. Results[0].Distance != 0 {
	        return errors.New("first result is not 0")
	}
```

Comment (rinx) [reply tests/e2e/operation/multi.go:54]: Thanks, but it should be like the following.
```suggestion
	res, err := client.MultiSearch(ctx, req)
	if err != nil {
		return err
	}
	
	if len(res.GetResponses()) != len(ds.Test) {
		return errors.New("number of responses does not match with sent requests")
	}
```

The first element of each result doesn't always have 0 distance.

Comment (kevindiu) [reply tests/e2e/operation/multi.go]: thank you. should we check the result contains 0 distance element?

Comment (rinx) [reply tests/e2e/operation/multi.go]: No. in almost cases, the results don't contain 0 distance element.

File: tests/e2e/operation/multi.go [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+		reqs = append(reqs, &payload.Search_IDRequest{
+			Id:     strconv.Itoa(i),
+			Config: cfg,
+		})
+	}
+
+	req := &payload.Search_MultiIDRequest{
+		Requests: reqs,
+	}
+	_, err = client.MultiSearchByID(ctx, req)
```
Comment (kevindiu) [inline tests/e2e/operation/multi.go:86]: should we check the length also? (all of the functions in this file)

Comment (kevindiu) [reply tests/e2e/operation/multi.go:86]: other than this comment, LGTM for me

Comment (rinx) [reply tests/e2e/operation/multi.go]: Thanks! I added them.

**Review by kevindiu — APPROVED** [review]
LGTM! thank you!

**Review by hlts2 — APPROVED** [review]
LGTM

**Pull request merged**